### PR TITLE
Allow white spaces to show so that one position text-based emojis better 😄 

### PR DESCRIPTION
**Pull request by jonico**
:wave: @crinjely,

first of all I like to thank you so much for your ✨ project. My daughter Sophie is a HUGE Roblox fan like you and is having her 10th birthday this weekend. She is designing all kind of T-shirts for her groups ([like this one](https://www.roblox.com/groups/9393178/Die-Blaupastelschmetterlingfashionistas#!/about)) and knows how time consuming this can be. Your app will be an amazing birthday present for her: https://roblox-444galaxygirl.herokuapp.com/about/

I would also like to give you something back for your app in the spirit of true Open Source: When trying out the text feature in your app, I noticed that white spaces at the beginning of a text get ignored. This makes it harder to position emojis in the center of the shirt. Sophie figured that emojis are often the easiest way to create fancy T-Shirts because [there are so many of them](https://getemoji.com/), and they scale with the font size:

image

If you merge this pull request, you will be able to position text based emojis using white spaces as well 🎉 

Thanks again for your wonderful project

:octocat: ❤️ 

/cc @assafweinberg thank you so much for introducing kids into tech, you may also like https://myoctocat.dev/

### CODE REVIEW

File: src/components/SelectionArea.js [hunk 1]
```diff
@@ -24,8 +24,9 @@ function getTextStyle(data) {
     const size = {fontSize: data.fontSize}
     const color = {color: data.textColor}
     const fontFamily = {fontFamily: data.fontFamily}
+    const showWhitespaces = { whiteSpace: "pre" }
 
-    return Object.assign({}, size, color, fontFamily)
+    return Object.assign({}, size, color, fontFamily, showWhitespaces)
```
Comment (crinjely) [inline src/components/SelectionArea.js:29]: Let's make that style object more generic. For example:

```
const baseStyles = { whiteSpace: "pre" }
return Object.assign(baseStyles, size, color, fontFamily)
```

**Review by crinjely — APPROVED** [review]
Thank you for helping with the project! My dad made a little suggestion, but it looks good!

**Comment by crinjely** [comment]
Happy birthday Sophie! Thanks for helping with the project and thanks for posting it with my picture. How long have you been learning to code? Do you have more ideas to add? Hope you have a great birthday!

File: src/components/SelectionArea.js (hunk 1 above)
Comment (jonico) [inline src/components/SelectionArea.js:29]:
```suggestion
    const baseStyles = { whiteSpace: "pre" }
    return Object.assign(baseStyles, size, color, fontFamily)
```

File: src/components/SelectionArea.js [hunk 2]
```diff
@@ -24,8 +24,9 @@ function getTextStyle(data) {
     const size = {fontSize: data.fontSize}
     const color = {color: data.textColor}
     const fontFamily = {fontFamily: data.fontFamily}
+    const showWhitespaces = { whiteSpace: "pre" }
```
Comment (jonico) [inline src/components/SelectionArea.js:27]:
```suggestion
    const baseStyles = { whiteSpace: "pre" }
    return Object.assign(baseStyles, size, color, fontFamily)
```

File: src/components/SelectionArea.js (hunk 1 above)
Comment (jonico) [reply src/components/SelectionArea.js:29]: great, idea, adding a follow up commit right now

**Comment by jonico** [comment]
> Happy birthday Sophie! Thanks for helping with the project and thanks for posting it with my picture. How long have you been learning to code? Do you have more ideas to add? Hope you have a great birthday!

Sophie's birthday is not before Sunday, so this is still a secret to her 😊 Will let you know once the big day arrives.

**Comment by jonico** [comment]
@crinjely - in the mean time, you can unwrap a small virtual present if you like to - just create an issue in [this GitHub project](https://github.com/winterfest-2020/unwrap-parcel) [using this link](https://github.com/winterfest-2020/unwrap-parcel/issues/new?assignees=&labels=parcel%2C+winterfest&template=send-encrypted-winterfest-parcel.md&title=Send+me+my+encrypted+Winterfest+parcel+please).

**Comment by jonico** [comment]
@crinjely: Sophie likes to thank you very much for her birthday surprise and is eagerly designing new 👚 right now:

image

image

**Pull request merged**